src/drivers/intel/cavs/dmic.c

```diff
@@ -1486,6 +1486,13 @@ static int dmic_remove(struct dai *dai)
 	rfree(dma_get_drvdata(dai));
 	dai_set_drvdata(dai, NULL);
 
+#if defined(CONFIG_CANNONLAKE) || defined(CONFIG_ICELAKE) \
+	|| defined(CONFIG_SUECREEK)
+	/* Disable DMIC power */
+	io_reg_write(DMICLCTL_REG,
```

Comment:
use pm_runtime_put() here, btw where is power enabled in dmic.c It should be with a call to pm_runtime_get_sync().